### PR TITLE
Replace String concatenation with Log4j ParameterizedMessage for readability

### LINKED ISSUE #905: Replace String concatenation with Log4j ParameterizedMessage for readability
### Is your feature request related to a problem?

Log4j includes parameter substitution which improves readability of log messages and has improved performance and stability over similar `String.format()`:
```
logger.info("Updating {} with {}.", baz.getFoo(), baz.getBar());
```
This works well when we are only logging a message.  However, we have many cases in our Transport Action exception handling where we use the same message in both the log and exception message (I know because I wrote them!).  One such example:
https://github.com/opensearch-project/flow-framework/blob/7a93d6c543738c7250313b0f66f3f794f42f5f1b/src/main/java/org/opensearch/flowframework/transport/ProvisionWorkflowTransportAction.java#L140-L144

Log4J's `ParameterizedMessage` class provides a way to construct the message the same way for both cases.

### What solution would you like?

```java
import org.apache.logging.log4j.message.ParameterizedMessageFactory;

} catch (Exception e) {
    String errorMessage = ParameterizedMessageFactory.INSTANCE.newMessage(
        "Failed to retrieve template from global context for workflow {}",
        workflowId
    ).getFormattedMessage();
    logger.error(errorMessage, e);
    listener.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(e)));
}
```

Example commit inspiring this issue: https://github.com/opensearch-project/flow-framework/pull/898/commits/86ac8ea862ba59abcbdf036fd637159e751e3be4

### What alternatives have you considered?

1. Keep the current concatenation (less readable)
2. Replace with `StringBuilder` implementation (less readable)
3. Replace with `String.format()` (performance, can throw exceptions, locale concerns)

### Do you have any additional context?

There are lots of these.  As a good first issue, doing any one of them, or perhaps a class at a time, would be a welcome contribution!

In addition, since most of them follow the same pattern, a new method (per class) of `handleException(Exception e, String formatString, Object... args)` would probably be useful.

### PR DESCRIPTION
**Pull request by KirrTap**
### Description
I have replaced string concatenation with Log4j ParameterizedMessage in 40 files.

### Related Issues
Resolves #905 

### Check List
- [x] Commits are signed per the DCO using `--signoff`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/flow-framework/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/flowframework/rest/RestCreateWorkflowAction.java [hunk 1]
```diff
@@ -242,7 +244,9 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
                     XContentBuilder exceptionBuilder = ex.toXContent(channel.newErrorBuilder(), ToXContent.EMPTY_PARAMS);
                     channel.sendResponse(new BytesRestResponse(ex.getRestStatus(), exceptionBuilder));
                 } catch (IOException e) {
-                    String errorMessage = "IOException: Failed to send back create workflow exception";
+                    String errorMessage = ParameterizedMessageFactory.INSTANCE.newMessage(
+                        "IOException: Failed to send back create workflow exception"
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/flowframework/rest/RestCreateWorkflowAction.java:248]: No need to rewrite this one (and many similar), EXCEPT... if you want, you could create a constant in the `CommonValue` class, something like `FAILED_TO_SEND_EXCEPTION` with the value "IOException: Failed to send back {} exception".  Then here you'd add the string "create workflow" and in the other similar ones you'd change it (get, delete, etc.).

File: src/main/java/org/opensearch/flowframework/rest/RestDeleteWorkflowAction.java [hunk 2]
```diff
@@ -92,7 +93,9 @@ protected BaseRestHandler.RestChannelConsumer prepareRequest(RestRequest request
                     XContentBuilder exceptionBuilder = ex.toXContent(channel.newErrorBuilder(), ToXContent.EMPTY_PARAMS);
                     channel.sendResponse(new BytesRestResponse(ex.getRestStatus(), exceptionBuilder));
                 } catch (IOException e) {
-                    String errorMessage = "IOException: Failed to send back delete workflow exception";
+                    String errorMessage = ParameterizedMessageFactory.INSTANCE.newMessage(
+                        "IOException: Failed to send back delete workflow exception"
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/flowframework/rest/RestDeleteWorkflowAction.java:97]: see above; no need to replace a plain string but consider parameterizing "delete"

File: src/main/java/org/opensearch/flowframework/rest/RestDeprovisionWorkflowAction.java [hunk 3]
```diff
@@ -94,7 +95,9 @@ protected BaseRestHandler.RestChannelConsumer prepareRequest(RestRequest request
                     XContentBuilder exceptionBuilder = ex.toXContent(channel.newErrorBuilder(), ToXContent.EMPTY_PARAMS);
                     channel.sendResponse(new BytesRestResponse(ex.getRestStatus(), exceptionBuilder));
                 } catch (IOException e) {
-                    String errorMessage = "IOException: Failed to send back deprovision workflow exception";
+                    String errorMessage = ParameterizedMessageFactory.INSTANCE.newMessage(
+                        "IOException: Failed to send back deprovision workflow exception"
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/flowframework/rest/RestDeprovisionWorkflowAction.java:99]: see above; no need to replace a plain string but consider parameterizing "deprovision workflow"

File: src/main/java/org/opensearch/flowframework/rest/RestGetWorkflowAction.java [hunk 4]
```diff
@@ -91,7 +92,9 @@ protected BaseRestHandler.RestChannelConsumer prepareRequest(RestRequest request
                     XContentBuilder exceptionBuilder = ex.toXContent(channel.newErrorBuilder(), ToXContent.EMPTY_PARAMS);
                     channel.sendResponse(new BytesRestResponse(ex.getRestStatus(), exceptionBuilder));
                 } catch (IOException e) {
-                    String errorMessage = "IOException: Failed to send back get workflow exception";
+                    String errorMessage = ParameterizedMessageFactory.INSTANCE.newMessage(
+                        "IOException: Failed to send back get workflow exception"
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/flowframework/rest/RestGetWorkflowAction.java:96]: see above; no need to replace a plain string but consider parameterizing "get workflow"

File: src/main/java/org/opensearch/flowframework/rest/RestGetWorkflowStateAction.java [hunk 5]
```diff
@@ -87,7 +88,9 @@ protected BaseRestHandler.RestChannelConsumer prepareRequest(RestRequest request
                     XContentBuilder exceptionBuilder = ex.toXContent(channel.newErrorBuilder(), ToXContent.EMPTY_PARAMS);
                     channel.sendResponse(new BytesRestResponse(ex.getRestStatus(), exceptionBuilder));
                 } catch (IOException e) {
-                    String errorMessage = "IOException: Failed to send back get workflow state exception";
+                    String errorMessage = ParameterizedMessageFactory.INSTANCE.newMessage(
+                        "IOException: Failed to send back get workflow state exception"
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/flowframework/rest/RestGetWorkflowStateAction.java:92]: see above; no need to replace a plain string but consider parameterizing "get workflow state"

File: src/main/java/org/opensearch/flowframework/rest/RestGetWorkflowStepAction.java [hunk 6]
```diff
@@ -91,7 +92,9 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
                     XContentBuilder exceptionBuilder = ex.toXContent(channel.newErrorBuilder(), ToXContent.EMPTY_PARAMS);
                     channel.sendResponse(new BytesRestResponse(ex.getRestStatus(), exceptionBuilder));
                 } catch (IOException e) {
-                    String errorMessage = "IOException: Failed to send back get workflow step exception";
+                    String errorMessage = ParameterizedMessageFactory.INSTANCE.newMessage(
+                        "IOException: Failed to send back get workflow step exception"
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/flowframework/rest/RestGetWorkflowStepAction.java:96]: see above; no need to replace a plain string but consider parameterizing "get workflow step"

File: src/main/java/org/opensearch/flowframework/rest/RestProvisionWorkflowAction.java [hunk 7]
```diff
@@ -98,7 +99,9 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
                     XContentBuilder exceptionBuilder = ex.toXContent(channel.newErrorBuilder(), ToXContent.EMPTY_PARAMS);
                     channel.sendResponse(new BytesRestResponse(ex.getRestStatus(), exceptionBuilder));
                 } catch (IOException e) {
-                    String errorMessage = "IOException: Failed to send back provision workflow exception";
+                    String errorMessage = ParameterizedMessageFactory.INSTANCE.newMessage(
+                        "IOException: Failed to send back provision workflow exception"
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/flowframework/rest/RestProvisionWorkflowAction.java:103]: see above; no need to replace a plain string but consider parameterizing "provision workflow"

File: src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java [hunk 8]
```diff
@@ -256,7 +263,9 @@ private void createExecute(WorkflowRequest request, User user, ActionListener<Wo
                                                         ActionListener.wrap(provisionResponse -> {
                                                             listener.onResponse(new WorkflowResponse(provisionResponse.getWorkflowId()));
                                                         }, exception -> {
-                                                            String errorMessage = "Provisioning failed.";
+                                                            String errorMessage = ParameterizedMessageFactory.INSTANCE.newMessage(
+                                                                "Provisioning failed."
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java:267]: plain string

File: src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java [hunk 9]
```diff
@@ -274,7 +283,9 @@ private void createExecute(WorkflowRequest request, User user, ActionListener<Wo
                                                     listener.onResponse(new WorkflowResponse(globalContextResponse.getId()));
                                                 }
                                             }, exception -> {
-                                                String errorMessage = "Failed to save workflow state";
+                                                String errorMessage = ParameterizedMessageFactory.INSTANCE.newMessage(
+                                                    "Failed to save workflow state"
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java:287]: plain string

**Review by dbwiddis — COMMENTED** [review]
Thanks for submitting this PR!

Appreciate all the work -- one thing to note is that you don't need to do anything with values that are already just plain strings with nothing to insert into them.  Just leave those alone.

Except, as indicated, when there are a lot of strings with similar text except for a few words, you might consider making a log message constant in the CommonValue class and using it in those places.

I stopped after several files, will let you submit an update before a more complete review.

**Review by dbwiddis — APPROVED** [review]
Thanks for this PR!  

One last request, please add an entry in CHANGELOG.md under the "Refactoring" header linking to this PR.  After that it should be ready to merge.

**Review by owaiskazi19 — APPROVED** [review]
Thanks @KirrTap for the change

**Comment by owaiskazi19** [comment]
@KirrTap can you do a manual backport to 2.x change and raise a PR?

**Pull request merged**